kolibri/plugins/learn/assets/test/views/learn-immersive-layout.spec.js

```diff
@@ -0,0 +1,57 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils';
+import makeStore from '../makeStore';
+import LearnImmersiveLayout from '../../src/views/LearnImmersiveLayout';
+
+const localVue = createLocalVue();
+
+const store = makeStore();
+store.state.core = {
+  blockDoubleClicks: true,
+};
+
+function makeWrapper({ propsData } = {}) {
+  return shallowMount(LearnImmersiveLayout, {
+    propsData,
+    store,
+    localVue,
+    stubs: {
+      LearningActivityBar: {
+        name: 'LearningActivityBar',
+        propsData: {
+          resourceTitle: 'Test Title',
+        },
+        template: '<div></div>',
+      },
+      LessonMasteryBar: {
+        name: 'LessonMasteryBar',
+        template: '<div></div>',
+      },
+      ContentPage: {
+        name: 'ContentPage',
+        template: '<div><slot></slot></div>',
+      },
+    },
+  });
+}
+
+describe('LearnImmersiveLayout', () => {
+  const wrapper = makeWrapper();
+  it('smoke test', () => {
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('shows the Learning Activity Bar', () => {
+    const wrapper = makeWrapper();
+    expect(wrapper.find('[data-test="learningActivityBar"]').exists()).toBeTruthy();
+  });
+
+  it('does not show the Mastery Bar by default', () => {
```

Comment:
It might be useful here to write a test for the case of when we do show the mastery bar

Reply:
Yes - that's a good idea 😄